### PR TITLE
Changed `Record` syntax

### PR DESCRIPTION
**Pull request by dansteren**
Records are now expected wrapped in a `Record` type, instead of just an object literal.

Example:

```ts
type SomeRecord = Record<{
  some_prop: string
}>
```

### CODE REVIEW

File: examples/inline_types/src/types.ts [hunk 1]
```diff
@@ -10,17 +10,18 @@ import {
...
     query,
+    Record,
     update,
     Variant
 } from 'azle';
 
-export type User1 = {
+export type User1 = Record<{
     id: string;
     job: {
```
Comment (lastmjs) [inline examples/inline_types/src/types.ts:20]: We should discuss the inline records having the `Record` wrapping as well, just like inline variants.

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/inline_types/src/types.ts [hunk 2]
```diff
@@ -10,17 +10,18 @@ import {
     nat,
     nat64,
     query,
+    Record,
```
Comment (lastmjs) [inline examples/inline_types/src/types.ts:13]: The parsing should break if the type is not wrapped in `Record`.

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/inline_types/src/types.ts (hunk 1 above)
Comment (dansteren) [reply examples/inline_types/src/types.ts]: I've updated the parser to handle/require this.

File: examples/inline_types/src/types.ts (hunk 2 above)
Comment (dansteren) [reply examples/inline_types/src/types.ts:13]: I've updated the parser to handle/require this.

File: canisters/ledger/index.ts [hunk 3]
```diff
@@ -221,39 +252,28 @@ export type QueryBlocksResponse = {
...
-        // The number of blocks that can be fetch using the callback.
-        length: nat64;
-
-        // The function that should be called to fetch the archived blocks.
-        // The range of the blocks accessible using this function is given by [from]
-        // and [len] fields above.
-        callback: QueryArchiveFn;
-    }[];
-};
+    archived_blocks: ArchivedBlock[];
```
Comment (lastmjs) [inline canisters/ledger/index.ts:255]: Let's revert these to inline types

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Review by lastmjs — APPROVED** [review]

File: canisters/ledger/index.ts (hunk 3 above)
Comment (dansteren) [reply canisters/ledger/index.ts]: Done

**Pull request merged**